### PR TITLE
Migrate to vue language server 2.x

### PR DESCRIPTION
**Pull request by rchl**
Migrating this package to use [Vue Language Server](https://github.com/vuejs/language-tools) as the underlying server (legacy Vetur-based server is available through https://packagecontrol.io/packages/LSP-vetur for those who really need it).

Note that by default hybrid mode is disabled which means that this server handles everything in the Vue file. Primarily due to https://github.com/sublimelsp/LSP/issues/2526 it's not recommended to enable hybrid mode.

The changes are relative to the current code of LSP-volar.

### CODE REVIEW

File: messages/3.0.0.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Breaking change
+
+This server is now based on the Vue Language Server 2.x rather than the deprecated and no longer updated Vetur.
+
+The legacy, deprecated Vetur is now available through the LSP-vetur package for those who really need it.
```
Comment (predragnikolic) [inline messages/3.0.0.txt:5]: Maybe extend the release notes with a guide on how the users should handle this breaking change.
For example, if they have a sublime-project file they would probably need to change the name from LSP-volar to LSP-vue.

File: plugin.py [hunk 2]
```diff
@@ -7,34 +8,36 @@
...
         window: sublime.Window,
-        initiating_view: Optional[sublime.View] = None,
-        workspace_folders: Optional[List[WorkspaceFolder]] = None,
-        configuration: Optional[ClientConfig] = None
+        initiating_view: sublime.View,
+        workspace_folders: List[WorkspaceFolder],
+        configuration: ClientConfig
     ) -> Optional[str]:
-        if not workspace_folders or not configuration:
-            return 'Can not run without a workspace folder'
```
Comment (predragnikolic) [inline plugin.py:37]: nice, so they added support for opening a single file

**Review by predragnikolic — APPROVED** [review]

File: LSP-vue.sublime-settings [hunk 3]
```diff
@@ -1,26 +1,10 @@
...
-		// By default all Volar trigger characters are disabled as ST does a better job.
-		"ignoreTriggerCharacters": [],
-		"json.customBlockSchemaUrls": {},
-		// Reverse priority for tsconfig pickup.
-		"reverseConfigFilePriority": false,
-		// Specify the `typescript.tsdk` to tsserverlibrary.js.
-		// If not specified LSP-volar will use the typescript from the workspace if it exist,
-		// else the typescript bundled with LSP-volar will be used.
-		// "typescript.tsdk": "/path/to/tsserverlibrary.js",
+		// DON'T use this option until related issues are ironed out - see https://github.com/sublimelsp/LSP/issues/2526
```
[the remark this answers is not in the document]
Comment (rchl) [reply LSP-vue.sublime-settings:3]: Some issues are still not ironed out but the link itself is outdated, yes. I will have to rephrase it.

**Comment by predragnikolic** [comment]
Do you also plan to add what you wrote here https://github.com/sublimelsp/LSP/issues/2526#issuecomment-2458086743 in this PR?

File: messages/3.0.0.txt (hunk 1 above)
Comment (rchl) [reply messages/3.0.0.txt:5]: Keep in mind that this message will show up for existing LSP-vue and not LSP-volar users. There is not much the LSP-vue user can do in terms of migration - either just start using new server or switch to LSP-vetur.

**Comment by rchl** [comment]
> Do you also plan to add what you wrote here [sublimelsp/LSP#2526 (comment)](https://github.com/sublimelsp/LSP/issues/2526#issuecomment-2458086743) in this PR?

I have in mind to make hybrid mode usable but for now it's advised not to use it.

File: messages/3.0.0.txt (hunk 1 above)
Comment (predragnikolic) [reply messages/3.0.0.txt:5]: Right, I got confused by the diff

**Review by predragnikolic — APPROVED** [review]

**Pull request merged**